SimTracker/TrackTriggerAssociation/plugins/TTStubAssociator.cc

```diff
@@ -29,33 +29,29 @@ void TTStubAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, const
 
   /// Loop over the InputTags to handle multiple collections
 
-  for (auto iTag = TTStubsTokens.begin(); iTag != TTStubsTokens.end(); iTag++) {
+  for (const auto& iTag : ttStubsTokens) {
     /// Prepare output
     auto associationMapForOutput = std::make_unique<TTStubAssociationMap<Ref_Phase2TrackerDigi_>>();
 
     /// Get the Stubs already stored away
-    edm::Handle<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>> TTStubHandle;
-    iEvent.getByToken(*iTag, TTStubHandle);
+    edm::Handle<TTStubDetSetVec> ttStubHandle;
+    iEvent.getByToken(iTag, ttStubHandle);
 
     /// Get the Cluster MC truth
-    edm::Handle<TTClusterAssociationMap<Ref_Phase2TrackerDigi_>> TTClusterAssociationMapHandle;
-    iEvent.getByToken(TTClusterTruthTokens.at(ncont1), TTClusterAssociationMapHandle);
+    edm::Handle<TTClusterAssociationMap<Ref_Phase2TrackerDigi_>> ttClusterAssociationMapHandle;
+    iEvent.getByToken(ttClusterTruthTokens.at(ncont1), ttClusterAssociationMapHandle);
 
     /// Prepare the necessary maps
-    std::map<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>, TTStub<Ref_Phase2TrackerDigi_>>,
-             edm::Ptr<TrackingParticle>>
-        stubToTrackingParticleMap;
-    std::map<edm::Ptr<TrackingParticle>,
-             std::vector<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>, TTStub<Ref_Phase2TrackerDigi_>>>>
-        trackingParticleToStubVectorMap;
+    std::map<TTStubRef, TrackingParticlePtr> stubToTrackingParticleMap;
+    std::map<TrackingParticlePtr, std::vector<TTStubRef>> trackingParticleToStubVectorMap;
     stubToTrackingParticleMap.clear();
```

Comment:
no need to clear maps right after they're created

Reply:
done